elastic/eui
# [Emotion] Convert EuiDataGrid cell popover, actions, and focus outline styles (Part 4)

## Summary

This PR:
- Converts cell popover, actions, and focus outline to Emotion (https://github.com/elastic/eui/issues/6394)
- Addresses some of https://github.com/elastic/eui/issues/7828#issuecomment-2176631041 (marked in individual commits)
- Fixes a bug with EuiDataGridCell not correctly merging `data-test-subj`s

As always, I recommend [following along by commit](https://github.com/elastic/eui/pull/8011/commits). The last commit/refactor is a bit spicy - I'd love to hear feedback on whether y'all feel it helps make the CSS more understandable & maintainable or if it does the opposite 🫠 

## QA

- Go to https://eui.elastic.co/pr_8011/#/tabular-content/data-grid or https://eui.elastic.co/pr_8011/storybook/?path=/story/tabular-content-euidatagrid--playground
  - [x] On hover (but not click), confirm that the cell outline+actions are gray and the cell actions have a slight delay before showing up
  - [x] On click, confirm that the cell outline+actions are blue, and cell actions show up immediately
  - [x] On keyboard arrow navigation of cells, confirm that cell actions show up immediately and outline+actions are always blue
- New hover zone affordance testing
  - [x] On hover (without clicking), move your mouse diagonally from the cell content to the actions and confirm that there is more buffer/affordance for this compared to production
- Go to your OS system accessibility settings and toggle `Reduce motion`
  - [x] Confirm cell actions show up immediately on hover
  - [x] Confirm cell popovers show with no transition/animation on open
- Header cell outline testing - go to [Storybook](https://eui.elastic.co/pr_8011/storybook/index.html?path=/story/tabular-content-euidatagrid--custom-header-content) or [focus docs](https://eui.elastic.co/pr_8011/storybook/#/tabular-content/data-grid-cells-popovers#focus)
  - [x] Confirm no outline shows on hover
  - [x] Confirm the focus outline shows on click
  - [x] Confirm the focus outline shows when the cell actions is opened both via keyboard enter and mouse click

### General checklist

- Browser QA
    ~- [ ] Checked in **mobile**~
- Docs site QA - N/A
- Code quality checklist
    - [x] Added **[visual regression tests](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/visual-regression-testing.md)**
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
